hive-metastore/src/main/java/org/apache/iceberg/hive/HiveHadoopUtil.java

```diff
@@ -31,7 +31,12 @@ private HiveHadoopUtil() {}
 
   public static String currentUser() {
     try {
-      return UserGroupInformation.getCurrentUser().getUserName();
+      String username = UserGroupInformation.getCurrentUser().getShortUserName();
+      if (username != null) {
+        return username;
+      } else {
+        throw new IOException("Hadoop user is null");
```

Comment:
I think it's fine the original way, else this is going to print a useless stack.  

Let's just put something like:
```
else {
   LOG.warn("Hadoop user is null, defaulting to user.name")
   return System.getProperty("user.name");
}
```

Reply:
OK